colibre/auto_plotter/gas_fraction_plots.yml

```diff
@@ -0,0 +1,183 @@
+h2_frac_func_stellar_mass:
+  type: "scatter"
+  legend_loc: "lower left"
+  selection_mask: "derived_quantities.xcoldgass_sel"
+  comment: r"$10^9 M_{\odot} < M_{\ast} < 10^{11.5} M_{\odot}$"
+  x:
+    quantity: "apertures.mass_star_30_kpc"
+    units: Solar_Mass
+    start: 1e9
+    end: 4e11
+  y:
+    quantity: "derived_quantities.h2_to_stellar_mass"
+    units: "Solar_Mass / Solar_Mass"
+    start: 1e-4
+    end: 100
+  median:
+    plot: true
+    log: true
+    number_of_bins: 9
```

Comment:
I don't fully understand why we would want an odd number of bins here. Is it worth expanding the plot limits to 1e9 -> 1e12, and have there be 15 equal-width bins of 0.2 dex here, consistent with the rest of the plots? We can then use the `adaptive: true` to use adaptive binning too.